hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -584,29 +583,28 @@ public Response getContainerMisMatchInsights(
         new ArrayList<>();
     Long minContainerID = prevKey + 1;
     Iterator<ContainerInfo> scmNonDeletedContainers =
-            containerManager.getContainers().stream()
-                    .filter(containerInfo -> (containerInfo.getContainerID() >= minContainerID))
-                    .filter(containerInfo -> containerInfo.getState() != HddsProtos.LifeCycleState.DELETED)
-                    .sorted(Comparator.comparingLong(ContainerInfo::getContainerID)).iterator();
-    ContainerInfo scmContainerInfo = scmNonDeletedContainers.hasNext() ?
-            scmNonDeletedContainers.next() : null;
+            containerManager.getContainerInfoIterator(ContainerID.valueOf(minContainerID),
+              containerInfo -> containerInfo.getState() != HddsProtos.LifeCycleState.DELETED);
+    ContainerInfo scmContainerInfo = scmNonDeletedContainers.hasNext() ? scmNonDeletedContainers.next() : null;
     DataFilter dataFilter = DataFilter.fromValue(missingIn.toUpperCase());
     try (SeekableIterator<Long, ContainerMetadata> omContainers =
-                 reconContainerMetadataManager.getContainersIterator()) {
+           reconContainerMetadataManager.getContainersIterator()) {
       omContainers.seek(minContainerID);
-      ContainerMetadata containerMetadata = omContainers.hasNext() ? omContainers.next() : null;
+
       switch (dataFilter) {
       case SCM:
         List<ContainerMetadata> notSCMContainers = new ArrayList<>();
-        while (containerMetadata != null && notSCMContainers.size() < limit) {
-          Long omContainerID = containerMetadata.getContainerID();
+        while (omContainers.hasNext() && notSCMContainers.size() < limit) {
+          Long omContainerID = omContainers.peekNextKey();
           Long scmContainerID = scmContainerInfo == null ? null : scmContainerInfo.getContainerID();
           if (omContainerID.equals(scmContainerID)) {
-            containerMetadata = omContainers.hasNext() ? omContainers.next() : null;
+            omContainers.seek(omContainerID + 1);
             scmContainerInfo = scmNonDeletedContainers.hasNext() ? scmNonDeletedContainers.next() : null;
           } else if (scmContainerID == null || omContainerID.compareTo(scmContainerID) < 0) {
-            notSCMContainers.add(containerMetadata);
-            containerMetadata = omContainers.hasNext() ? omContainers.next() : null;
+            ContainerMetadata omContainer = omContainers.next();
+            if (omContainer.getNumberOfKeys() > 0) {
+              notSCMContainers.add(omContainers.next());
```

Comment:
Is it correct to skip the container in `omContainers`, as we are calling .next() twice in same iteration ? Is it intentional ?

Reply:
Yeah I too found this bug. This is a miss on my part.